operators/pkg/instance-creation/exposition_test.go

```diff
@@ -0,0 +1,174 @@
+)
+
+func TestForgeService(t *testing.T) {
+	var (
+		name      = "usertest"
+		namespace = "namespacetest"
+	)
+
+	ownerRef := []metav1.OwnerReference{{
+		APIVersion: "crownlabs.polito.it/v1alpha2",
+		Kind:       "Instance",
+		Name:       "Test1",
+	},
+	}
+
+	service := ForgeService(name, namespace, ownerRef)
+
+	assert.Equal(t, service.ObjectMeta.Name, name+"-svc")
+	assert.Equal(t, service.ObjectMeta.Namespace, namespace)
+	assert.Equal(t, service.OwnerReferences[0].APIVersion, "crownlabs.polito.it/v1alpha2")
+	assert.Equal(t, service.OwnerReferences[0].Kind, "Instance")
+	assert.Equal(t, service.OwnerReferences[0].Name, "Test1")
+}
+
+func TestForgeIngress(t *testing.T) {
+	var (
+		name           = "usertest"
+		namespace      = "namespacetest"
+		urlUUID        = "urlUUIDtest"
+		websiteBaseUrl = "websiteBaseUrlTest"
+		svc            = corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "svc-test",
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						TargetPort: intstr.IntOrString{IntVal: 22},
+					},
+				},
+			},
+		}
+		url = websiteBaseUrl + "/" + urlUUID
+	)
+
+	ownerRef := []metav1.OwnerReference{{
+		APIVersion: "crownlabs.polito.it/v1alpha2",
+		Kind:       "Instance",
+		Name:       "Test1",
+	},
+	}
+
+	ingress := ForgeIngress(name, namespace, svc, urlUUID, websiteBaseUrl, ownerRef)
+
+	assert.Equal(t, ingress.ObjectMeta.Name, name+"-ingress")
+	assert.Equal(t, ingress.ObjectMeta.Namespace, namespace)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Name, svc.Name)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Port.Number, svc.Spec.Ports[0].TargetPort.IntVal)
+	assert.Equal(t, ingress.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/auth-signin"], "https://$host/"+urlUUID+"/oauth2/start?rd=$escaped_request_uri")
+	assert.Equal(t, ingress.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/auth-url"], "https://$host/"+urlUUID+"/oauth2/auth")
+	assert.Equal(t, ingress.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/configuration-snippet"], `sub_filter '<head>' '<head> <base href="https://$host/`+urlUUID+`/index.html">';`)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Path, "/"+urlUUID+"(/|$)(.*)")
+	assert.Equal(t, ingress.ObjectMeta.Annotations["crownlabs.polito.it/probe-url"], "https://"+url)
+	assert.Equal(t, ingress.Spec.TLS[0].Hosts[0], websiteBaseUrl)
+	assert.Equal(t, ingress.Spec.Rules[0].Host, websiteBaseUrl)
+	assert.Equal(t, ingress.OwnerReferences[0].APIVersion, "crownlabs.polito.it/v1alpha2")
+	assert.Equal(t, ingress.OwnerReferences[0].Kind, "Instance")
+	assert.Equal(t, ingress.OwnerReferences[0].Name, "Test1")
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Name, svc.Name)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Port.Number, svc.Spec.Ports[0].TargetPort.IntVal)
```

Comment:
These tests seem duplicated to me. Please order the checks to prevent this issue

Reply:
You are right, thank you